crates/sui-framework/packages/sui-framework/sources/k_of_n_upgrade_policy.move

```diff
@@ -0,0 +1,511 @@
+    }
+
+    /// A succesful upgrade.
+    struct UpgradePerformed has copy, drop {
+        /// the instance of the k out of n policy
+        upgrade_cap: ID,
+        /// the ID of the proposal (`ProposedUpgrade` instance)
+        proposal: ID,
+        /// digest of the proposal
+        digest: vector<u8>,
+        /// signer of the upgrade
+        signer: address,
+    }
+
+    /// A discarded upgrade.
+    struct UpgradeDiscarded has copy, drop {
+        /// the instance of the k out of n policy
+        upgrade_cap: ID,
+        /// the ID of the proposal (`ProposedUpgrade` instance)
+        proposal: ID,
+        /// digest of the proposal
+        digest: vector<u8>,
+        /// signer of the discarded upgrade transaction
+        signer: address,
+    }
+
+    /// Record the transfer for a `Ballot`.
+    struct BallotTransfered has copy, drop {
+        /// the ID of the proposal (`ProposedUpgrade` instance)
+        proposal: ID,
+        /// digest of the proposal
+        digest: vector<u8>,
+        /// ballot used for this vote
+        ballot: ID,
+        /// signer of the transfer operation
+        signer: address,
+        /// original owner of the ballot, that is the address the
+        /// ballot was first sent to
+        original_owner: address,
+        /// the receiver of the ballot
+        ballot_receiver: address,
+    }
+
+    /// Record a deleted `Ballot`.
+    struct BallotDeleted has copy, drop {
+        /// the ID of the proposal (`ProposedUpgrade` instance)
+        proposal: ID,
+        /// digest of the proposal
+        digest: vector<u8>,
+        /// ballot used for this vote
+        ballot: ID,
+        /// signer of the deleted operation
+        signer: address,
+        /// original owner of the ballot, that is the address the
+        /// ballot was first sent to
+        original_owner: address,
+    }
+
+    /// Allowed voters must in the [2, 100] range.
+    const EAllowedVotersError: u64 = 0;
+    /// Required votes must be less than allowed voters.
+    const ERequiredVotesError: u64 = 1;
+    /// The `Ballot` used to vote is not for the correct proposal (`ProposedUpgrade`)
+    const EInvalidBallot: u64 = 2;
+    /// Not enough votes to perform the upgrade.
+    const ENotEnoughVotes: u64 = 3;
+    /// An upgrade was issued already, and the operation requested failed.
+    const EAlreadyIssued: u64 = 4;
+    /// The operation required the signer to be the same as the upgrade proposer.
+    const ESignerMismatch: u64 = 5;
+    /// Proposal (`KofNUpgradeCap`) and upgrade (`ProposedUpgrade`) do not match.
+    const EInvalidProposalForUpgrade: u64 = 6;
+
+    /// Create a `KofNUpgradeCap` given an `UpgradeCap`.
+    /// The returned instance is the only and exclusive controller of upgrades. 
+    /// If the transaction sender is not in the set of allowed voters it will be
+    /// added. The sender is the publisher of the package which is always allowed
+    /// to vote.
```

Comment:
I don't believe this statement is correct -- As far as I can see there is only the requirement that the `voters` size is > 1, but not that the `tx_context::sender` is present in that set. 

I'd be in favor of not adding this check either, but more just calling out that this part of the comment may want to be removed.

Reply:
fair I changed the code but not the comment, thanks